### ISSUE TITLE
Submission for SC consideration: PEP 657 -- Include Fine Grained Error Locations in Tracebacks

### ISSUE THREAD

**Issue by isidentical** [python/steering-council#67]
I am submitting PEP 657 in behalf of it's authors after multiple rounds ([one](https://mail.python.org/archives/list/python-dev@python.org/thread/DB3RTYBF2BXTY6ZHP3Z4DXCRWPJIQUFD), [two](https://discuss.python.org/t/pep-657-include-fine-grained-error-locations-in-tracebacks/8629/23) and [three](https://mail.python.org/archives/list/python-dev@python.org/thread/KR7ACFCUNMHT4M7R4XNHGRFV27HZBDFD/)) of discussions. We have evaluated the possible options regarding the implementation, and updated the PEP to include those feedbacks and added the ones that doesn't fit to the rejected ideas section. We believe that maintenance is an important factor, and after actively testing other options (e.g only storing node_ids and linking them at the runtime) we decided that the current solution would be the most straight forward one in terms of the cost / benefit / usability / maintainability.  

Thanks

**Comment by pablogsal** [comment]
I'm adding this to our agenda. As I am a co-author, is up the rest of the SC to decide when and how to discuss this PEP :)

**Comment by warsaw** [comment]
There is one thing I don't understand from reading the PEP, regarding opt-out.  Is it possible to (easily, i.e. without replacing the except hook) to opt out of the more detailed traceback display even if the pyc files have the additional information?  I have an opinion about what it should do, but it's not clear to me from the PEP what the actual proposal is.

**Comment by pablogsal** [comment]
> There is one thing I don't understand from reading the PEP, regarding opt-out. Is it possible to (easily, i.e. without replacing the except hook) to opt out of the more detailed traceback display even if the pyc files have the additional information?

Yeah, we can complete this better, but the idea is that if `PYNODEBUGRANGES` is set, then they won't be displayed even if the pyc files have them

**Comment by warsaw** [comment]
Thanks!  The behavior is great; I would just suggest tightening up the text of the PEP around that issue.
